# The PID constants are applied to positional errors in the map frame
position_p_x: 1.0
position_p_y: 1.0
position_p_z: 3.0

# PID values are positive for forward acting PID loops and negative for
# reverse acting PID loops. P, I, and D should all be the same sign.
throttle_p: 1.5
throttle_i: 0.2
throttle_d: 0.4
throttle_accumulator_enable_threshold: 10.0
throttle_accumulator_max: 5.0
throttle_accumulator_min: -5.0

model_mass: 1.0

# Thrust levels are in m/s^2
min_thrust: 5.0
max_thrust: 100.0

battery_timeout: 1.0

xy_mixer: "4dof"

pitch_p: 1.5
pitch_i: 0.3
pitch_d: 0.2
pitch_accumulator_max: 5.0
pitch_accumulator_min: -5.0
pitch_accumulator_enable_threshold: 10.0

roll_p: 1.5
roll_i: 0.3
roll_d: 0.2
roll_accumulator_max: 5.0
roll_accumulator_min: -5.0
roll_accumulator_enable_threshold: 10.0

yaw_p: 0.2

min_side_thrust: 0.1
max_side_thrust: 10.0

throttle_max      : 1.0
throttle_min      : 0.0
throttle_max_rate : 200.0

pitch_max         : 0.3
pitch_min         : -0.3
pitch_max_rate    : 2.0

roll_max          : 0.3
roll_min          : -0.3
roll_max_rate     : 2.0

yaw_max           : 0.5
yaw_min           : -0.5
yaw_max_rate      : 10.0

# Height below which the drone is required to be level
level_flight_required_height: 0.15
# Height above the level flight height at which to allow the drone
# to begin tilting again
level_flight_required_hysteresis: 0.10

update_frequency: 60.0
startup_timeout: 15.0
update_timeout: 0.5

# Time to delay after arming
post_arm_delay: 0.2

# Time to take to ramp the throttle to hover throttle
takeoff_throttle_ramp_duration: 0.5

# Rate at which to descend during first phase
# must be negative
descend_rate: -0.7
# Rate at which to descend during last phase
# must be negative
cushion_rate: -0.35

# Rate at which to accelerate to descent velocity
# must be negative
descend_acceleration: -1.0
# Rate at which to accelerate to the final cushioned velocity
# must be positive
cushion_acceleration: 0.3

#height at which the contact switches must be unpressed
takeoff_max_height_switch_pressed: 0.25

# Height below which is considered landed
landing_detected_height: 0.2
